src/services/dashboards/widgets/_components/WidgetDataTable.vue

```diff
@@ -1,79 +1,97 @@
+                            :key="`th-${contextKey}-${fieldColIndex}`"
+                            :style="{
+                                minWidth: field.width || undefined,
+                                width: field.width || undefined,
+                            }"
                         >
-                            <span class="name"
-                                  :style="{ color: labelColorFormatter(index) }"
+                            <slot :name="`th-${field.name}`"
+                                  v-bind="getHeadSlotProps(field, fieldColIndex)"
                             >
-                                {{ labelTextFormatter(index) }}
-                            </span>
-                        </slot>
-                    </template>
-                    <template v-else-if="typeof value === 'string'">
-                        {{ value }}
-                    </template>
-                    <template v-else-if="typeof value === 'number'">
-                        <template v-if="type === 'size'">
-                            {{ byteFormatter(value) }}
-                        </template>
-                        <template v-else-if="type === 'number'">
-                            {{ numberFormatter(value) }}
-                        </template>
-                        <template v-else>
-                            {{ CURRENCY_SYMBOL[currency] }}{{ currencyMoneyFormatter(value, currency, currencyRates, true) }}
-                        </template>
-                    </template>
-                    <template v-else>
-                        --
-                    </template>
-                    <!--                    <template v-else>-->
-                    <!--                        <span v-if="value.isRaised" :class="{raised: value.isRaised}">-->
-                    <!--                            <span>{{ value.value }}</span>-->
-                    <!--                            <p-i name="ic_bold-arrow-up" width="0.75rem" />-->
-                    <!--                        </span>-->
-                    <!--                        <span>{{ value.value }}</span>-->
-                    <!--                    </template>-->
-                </div>
-            </template>
-        </p-data-table>
+                                <span class="th-contents"
+                                      :class="{
+                                          [field.textAlign || DATA_TABLE_CELL_TEXT_ALIGN.left]: true,
+                                          'has-icon': field.tooltipText,
+                                      }"
+                                >
+                                    <span class="th-text">
+                                        <slot name="th-format"
+                                              v-bind="getHeadSlotProps(field, fieldColIndex)"
+                                        >
+                                            <slot :name="`th-${field.name}-format`"
```

Comment:
suggestion: remove format postfix, remove named th slot

1. History: format postfix in data table component was to distinguish th slot and th format(with style) slot. But th slot was removed someday, so now, th-xxx-format exists only. 
=> I don't think format postfix is necessary.
2. I don't think both th and th-format slot are necessary, because developers can use `th` slot with `if` statement to use for the particular field.